builtin/credential/approle/path_role.go

```diff
@@ -521,8 +521,26 @@ func (b *backend) pathRoleSecretIDList(req *logical.Request, data *framework.Fie
 	return logical.ListResponse(listItems), nil
 }
 
-// setRoleEntry grabs a write lock and stores the options on an role into the storage.
-// Also creates a reverse index from the role's RoleID to the role itself.
+// validRoleConstraints tells if the role has at least one constraint enabled
+func validRoleConstraints(role *roleStorageEntry) (bool, error) {
```

Comment:
My main issue here is that it either always returns either `true, nil` or `false, <some error>` which means we only need one value. I think you should either remove the bool or the error, probably the bool, and it's simply valid if err is not nil. We can add in more complex logic later if needed.